### PR TITLE
Asset summary tab

### LINKED ISSUE #2984: Asset Detail: Summary tab
Part of epic #2896 

## Is your feature request related to a problem? Please describe 👀
[Figma designs](https://www.figma.com/file/rIK2zVFUVhtbuXo0RBzOzw/0033-2305_omSupply-ColdChain-UI?type=design&node-id=2653-16424&mode=design&t=8KRUuAo7s1V0r0RM-0)

image

### Moneyworks Jobcode 🧰

`OMSCC:AST`

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #2984

# 👩🏻‍💻 What does this PR do? 
 
image

This connects the update functionality to the asset, including
* Save button enables only when there's a change made
* A confirmation prompt is shown when you click save
* A confirmation prompt is shown if you navigate away from an edited page
* Validation is shown for the date fields (but this doesn't affect the save button validity)
* Error toast is displayed if there was a problem
* Success toast is save was ok

#3193 has been created to handle the location editing
The Status display and update I've left, will need another issue for that one.

# 🧪 How has/should this change been tested? 

## 📃 Documentation

- [ ] Part of the asset editing

### CODE REVIEW

**Comment by adamdewey** [comment]
I think we might need to include the asset code in the Asset Identification section (as well as in the breadcrumbs at the top of the page)?

Perhaps at the start of the section?

image

Just imagining that a lot of places will likely be using manual stickers with human readable asset codes to identify assets

**Review by jmbrunskill — APPROVED** [review]
LGTM, I notice we don't show the name anywhere on the summary page, I wonder if we should? Or maybe we don't need name and only a code? Question for Richard I guess...

We also need somewhere we can assign a store? Or maybe we should default to the current store, and only not set the store on central server?

**Comment by adamdewey** [comment]
> We also need somewhere we can assign a store? Or maybe we should default to the current store, and only not set the store on central server?

@jmbrunskill , @richardmoizeau , @mark-prins 

I think this highlights that there could be 2 big models of user functionality for asset management - centralised and decentralised. 

UNICEF are very much pushing a decentralised model for CCE - they want facilities to be able to capture data about their own assets on site, and they want to make it as easy as possible for the users, to the point that they want them to scan a barcode rather than enter any manual data themselves (of course would still need option to enter manual data for assets that don't have barcodes). 

For those local workers it wouldn't make sense to be able to choose a store as they are only concerned with their own facility. 

However for a more decentralised model, where an authority figure is assigning assets to multiple facilities then they would need a mechanism to be able to select which store the asset is in. 
Technically if they have all Store permissions they could log into each store and add the assets that way, but could be a bit tedious - would think they would prefer some kind of spreadsheet import functionality at a central level that then gets synced down to individual Stores?

**Comment by adamdewey** [comment]
> LGTM, I notice we don't show the name anywhere on the summary page, I wonder if we should? Or maybe we don't need name and only a code? Question for Richard I guess...

In my (limited) experience with working with assets at companies I don't recall a 'name' being associated with an asset, just a code, type and location - but it could be useful to have a notes section somewhere where some more human info could be stored? 

E.g. `Fridge with the blue sticker - behind the vaccine shelves in the lab. Fred currently using it to store insulin 03/24`

**Comment by adamdewey** [comment]
Interestingly the WHO IGA tool did include a name for assets, but you can see from this sample data that no-one ever used it...

image

**Comment by mark-prins** [comment]
So to summarise, this is what I'm thinking. sound ok?
- the store which an asset belongs to is the current store / when creating an asset it is assigned to the current store and the user doesn't have an option to change it
- if you are editing the asset while on the central server though, you can change the store
- you can allocate a store when importing from a spreadsheet, this allows the multi-store admin to set up a number of assets & stores easily
- the 'name' field should be renamed to 'notes' and added to the summary page (RHS perhaps, under the status). multi-line input field

**Comment by richardmoizeau** [comment]
All good for me !

**Comment by adamdewey** [comment]
Nice @mark-prins 

The only thing I'm wondering about - is how would this work: 

> if you are editing the asset while on the central server though, you can change the store

Will there be a special 'central asset management' Store where you log in and can see all assets? 

Because I'm just thinking - if you already have to be in a store to do the editing, what happens when you change the store of the asset, does it disappear when you refresh the page?

**Comment by jmbrunskill** [comment]
> Will there be a special 'central asset management' Store where you log in and can see all assets?

I think their maybe should be a central management store, which you need to login to edit things like asset cataglogues, pack variants etc. Right not we're relying on being on the central server, but I think having a dedicated store makes things clearer for the user. I think it needs a wider discussion though. This is a great example.

**Comment by adamdewey** [comment]
Yeah I agree. 

Although I do think there is a significant difference in use case between centrally managing Master Data for stock inventory / asset catalogues and centrally managing asset distribution. 

Ideally asset distribution would have a targeted vertical, in the same way that we have figured out the vertical for stock inventory distribution. 

An efficient asset distribution vertical would eliminate the need for the distribution to be centrally administered (conversely you _could_ say stock inventory distribution could be centrally administered, but we would now think that's ridiculous)

I can see there is frustration in the community about how difficult it is to manage their asset distribution - we have a chance to improve things here. 

Long story short: if (when) we have a good vertical mapped out for asset distribution it wouldn't make sense to have an editable (or even viewable) Store value in the Asset Summary / Detail pages for users operating Open mSupply inside a Store (the users would only care about their own assets which they know is in their current Store). 

I'll bring this up in the Telegram chat to discuss further!
